app/services/dqt_api_access.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+require "net/http"
+require "jwt"
+
+class DqtApiAccess
+  class << self
+    def token
+      if @token.nil? || token_about_to_expire?
+        request = Net::HTTP::Get.new(uri)
+
+        data = {
+          grant_type: grant_type,
+          scope: scope,
+          client_id: client_id,
+          client_secret: client_secret,
+        }
+
+        request.set_form_data(data)
+
+        response = Net::HTTP.start(uri.hostname, uri.port, use_ssl: use_ssl?, read_timeout: 20) do |http|
+          http.request(request)
+        end
+
+        if response.code == "200"
+          @token = JSON.parse(response.body)["access_token"]
+        else
+          raise "DQT access token could not be fetched"
```

Comment:
Would we want to raise in this situation? Or fail the validation?

Reply:
I would like to see it in sentry personally

Reply:
I would of thought we'd want to raise. Any async stuff can be re-queued. Any synchronous calls should know there was an error so we can look into it.

If we fail the validation silently because of technical glitch I feel that would have potentially a more negative impact.

I'm happy to raise a more specific error here and then decide what we want to do in the client

Reply:
ok, this now explicitly sends to sentry, also a custom error is raise for finer grain control later for the caller